### DISCUSSION TITLE
Conditional sync

### DISCUSSION THREAD

**Discussion by brondum** [fluxcd/flux2#733] [Q&A] [answered]
Hi.

Is there to make a deployment (kustomize or helm) conditional.
The scenario being that if you have multiple clusters of the same type, like 3 cluster looking at the same folder, but only the two of them actually needs an application.

We manage multiple cluster environments for multiple teams, which would give quite a lot of duplicate code to maintain if we have to put application specs into separate cluster folders. 

Any insights?

**Comment by kingdonb** [comment, marked as answer]
There is no conditional sync, but instead you can have your kustomizations for each environment selectively include resources from only the bases that you want in each environment. If two environments have something in common between them, then it goes in a base, and they can both reference the same base. If one environment has a special-case need, (or if only certain environments need that), then it goes in a different base and only those environments should reference that base.

This still has the effect of enabling you to avoid duplicating the same resources across environments. Does this make sense?

**Reply by brondum** [reply]
I think i somewhat get this. But can you provide an example of how you would structure this ?
This would still need a specific folder for each cluster, correct ? or i might be missing something?

**Reply by kingdonb** [reply]
In the example https://github.com/fluxcd/flux2-kustomize-helm-example there are separate folders for each cluster, in the example there is a shared "infrastructure" definition that is used for both clusters, and "apps" is separated into two lists, one for each cluster.

Both apps repositories inherit from a common `base/podinfo`, which is modified with different values for different hostnames served by each environment. The "podinfo" app definition is in common, but the values are per-cluster.

Hope this helps!

**Reply by brondum** [reply]
Thank you very much :)
